### PR TITLE
Fixes #23966: Update the rudderc technique json schema to match the minimal SchemaStore expectation

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23966

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

File: policies/rudderc/src/technique.schema.json [hunk 1]
```diff
@@ -55,40 +108,59 @@
...
+          "description": "Technique parameter description",
+          "markdownDescription": "```Optional```\n\nTechnique parameter description",
           "type": "string"
         },
         "documentation": {
+          "title": "technique parameter documentation",
           "type": "string",
-          "format": "markdown"
+          "description": "Technique parameter documentation, nor in Markdown format",
+          "markdownDescription": "```Optional```\n\nTechnique parameter documentation, not in markdown format"
```
Comment (amousset) [inline policies/rudderc/src/technique.schema.json:150]:
```suggestion
          "markdownDescription": "```Optional```\n\nTechnique parameter documentation, not in Markdown format"
```

**Comment by Fdall** [comment]
PR rebased

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
OK, squash merging this PR

**Comment by Fdall** [comment]
OK, merging this PR

**Pull request merged**